bootstore/src/schemes/v0/request_manager.rs

```diff
@@ -0,0 +1,401 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A mechanism for tracking in flight requests
+
+use super::{Config, Envelope, Msg, Request, RequestType, Share, SharePkg};
+use sled_hardware::Baseboard;
+use std::collections::{BTreeMap, BTreeSet};
+use std::time::Instant;
+use uuid::Uuid;
+
+/// Acknowledgement tracking for `RequestType::InitRack`.
+#[derive(Debug, Default)]
+pub struct InitAcks {
+    pub expected: BTreeSet<Baseboard>,
+    pub received: BTreeSet<Baseboard>,
+}
+
+/// Acknowledgement tracking for `RequestType::LoadRackSecret` and
+/// `RequestType::Learn`
+#[derive(Debug)]
+pub struct ShareAcks {
+    pub threshold: u8,
+    pub received: BTreeMap<Baseboard, Share>,
+}
+
+impl ShareAcks {
+    pub fn new(threshold: u8) -> ShareAcks {
+        ShareAcks { threshold, received: BTreeMap::new() }
+    }
+}
+
+/// A mechanism to track in flight requests
+///
+/// Manages expiry, acknowledgment, and retries
```

Comment:
This comment seems incorrect; I don't see anything about expiry / acknowledgement / retries in this `struct`